### DISCUSSION TITLE
To see incremental backup or full backup

### DISCUSSION THREAD

**Discussion by sikishen** [vmware-tanzu/velero#6000] [Community support Q&A]
Hi,

Is there a way to see whether the backup is incremental or full? i'm testing using command velero backup --snapshot-volumes --include-namespace, currently i'm not sure it's doing incremental or full, currently i can know is if at least one namespace backup TTL isn't expired i could still recovery from first backup point although TTL of that backup point already expired.

**Comment by blackpiglet** [comment]
Not sure about what's your scenario.
Did you create a schedule that triggers backup creation? Then need to clarify whether the latter backup was incremental based on the previous backup.

**Comment by sikishen** [comment]
Thanks for reply,
yes i created a schedule backup:
```
velero schedule create schedule-velero-test-schedule --schedule="@every 6h" --default-volumes-to-restic --include-namespaces schedule-velero-test --ttl 12h --snapshot-volumes
```
and after a weekend, first schedule backup already expired, but i could still recover from begining, which is a good thing, but i don't know whether it's doing incremental or full backup everytime, looks like it's doing full, but i want to have incremental backup stratgy using restic and i can find any option for it.

kubectl exec -n schedule-velero-test $(kubectl get pods -n schedule-velero-test -l "app=flog" -o name) -c ubuntu -- bash -c 'head -n 1 /var/log/generated.log;tail -n 1 /var/log/generated.log'

```
# log from start, start date is 3-17
{"host":"5.213.244.51", "user-identifier":"wintheiser5566", "datetime":"17/Mar/2023:06:46:16 +0000", "method": "POST", "request": "/aggregate/users/viral", "protocol":"HTTP/2.0", "status":504, "bytes":19874, "referer": "http://www.forwardbrand.biz/web-enabled/proactive"}
# log after a weekend, end date is 3-20
{"host":"93.140.249.113", "user-identifier":"jewess7120", "datetime":"20/Mar/2023:06:48:36 +0000", "method": "PATCH", "request": "/engineer/productize/proactive/harness", "protocol":"HTTP/1.1", "status":504, "bytes":16463, "referer": "http://www.forwardstreamline.org/virtual/e-markets/convergence"}
```

**Comment by blackpiglet** [comment]
The included k8s resources are backed up integral. Each backup triggered by the schedule should contain the same resources if the cluster has no change.
The incremental concept is for the data in the volume, when data mover is usage, e.g. Restic.

**Comment by sikishen** [comment]
Yes, i want to know about volume backup, i couldn't see whether data volume is backing up by incremental or full, since we have very large cluster needs to do migration, i want to make sure after inital backup finishes rest backup by schedule or manual are incremental, which i couldn't find any where.

I made another test:
kubectl exec -it -n incremental-test $(kubectl get pods -n incremental-test -l "app=storage-snapshot" -o name) bash
```
dd if=/dev/zero bs=1024 count=100000000 of=/mnt/100Gb.file
```
time velero backup create incremental-test-1 --include-namespaces incremental-test --snapshot-volumes --default-volumes-to-restic -w
```
# 100GB
real    16m25.603s
```
kubectl exec -it -n incremental-test $(kubectl get pods -n incremental-test -l "app=storage-snapshot" -o name) bash
```
dd if=/dev/zero bs=1024 count=50000000 of=/mnt/50Gb.file
```
time velero backup create incremental-test-2 --include-namespaces incremental-test --snapshot-volumes --default-volumes-to-restic -w
```
# 100GB(old) + 50GB(new), from backup time, seems it's doing incremental 
real    8m22.987s
```
is there a way that i could see that it's doing incremental, velero describe backup command doesn't show any info.

**Comment by blackpiglet** [comment]
When using FS uploader, the data is backed up incrementally.
Both Resitc and Kopia have the ability to de-duplicate, so the data stored in the OSS is handled properly.
When the old backup has reached the TTL, the data in OSS is only referenced by the old backup got pruned periodically.
If the data is still referenced by valid backups, the data will be kept. So the expired backup will not impact the data integrity.

**Comment by sikishen** [comment]
so as long as there's an available backup there, i could have a full restore from it, thank you for your answering!
